controllers/userController.js

```diff
@@ -175,17 +176,120 @@ const currentUser = async (req, res, next) => {
   ApiResponse.result(res, { user: data }, httpStatusCodes.OK);
 };
 
+const updateUser = async (req, res, next) => {
+  const { firstName, lastName, phone } = req.body;
+
+  let updateFields = {};
+
+  // Add fields to the update object if they are defined
+  if (firstName !== undefined) updateFields.firstName = firstName;
+  if (lastName !== undefined) updateFields.lastName = lastName;
+  if (phone !== undefined) updateFields.phone = phone;
+
+    const user = await Users.updateOne(
+      { _id: req.params.objectId },
+      { $set: updateFields }
+    );
+
+    ApiResponse.result(res, { user }, httpStatusCodes.OK);
+};
+
 //logout user
 const logoutUser = (req, res, next) => {
   res.cookie("accessToken", "", { maxAge: 0 });
   ApiResponse.result(res, { status: "User logged out successfully" }, httpStatusCodes.OK);
 };
+const changePassword = async (req, res, next) => {
+  const { oldpassword, newpassword } = req.body;
+  const objectId = req.user._userInfo
+   if (!oldpassword || !newpassword ){
+    throw new ApiError(
+      httpStatusCodes.BAD_REQUEST,
+      "All fields are required",
+      httpStatusCodes.BAD_REQUEST
+    );
+   }
+
+   console.log('user info ', objectId)
+   
+   const user = await Users.findOne({ _id: objectId });
```

Comment:
what is objectId here ? Can you see from my previous implementation find on basis of _id ? or you can search online

Reply:
i have extracted objectId from _userInfo
const objectId = req.user._userInfo

Reply:
No, do it like this:
new Types.ObjectId(req.user._userInfo.id)

import is like this
import { Types } from "mongoose";